tracim/tracim/lib/content.py

```diff
@@ -860,6 +860,57 @@ def move(self, item: Content,
 
         item.revision_type = ActionDescription.MOVE
 
+    def copy(
+        self,
+        item: Content,
+        new_parent: Content=None,
+        new_label: str=None,
+        do_save: bool=True,
+        do_notify: bool=True,
+    ) -> Content:
+        """
+        Copy nearly all content, revision included. Children not included, see
+        "copy_children" for this.
+        :param item: Item to copy
+        :param new_parent: new parent of the new copied item
+        :param new_label: new label of the new copied item
+        :param do_notify: notify copy or not
+        :return: Newly copied item
+        """
+        if (not new_parent and not new_label) or (new_parent == item.parent and new_label == item.label):  # nopep8
+            # TODO - G.M - 08-03-2018 - Use something else than value error
+            raise ValueError("You can't copy file into itself")
+        if new_parent:
+            workspace = new_parent.workspace
+            parent = new_parent
+        else:
+            workspace = item.workspace
+            parent = item.parent
+
+        if new_label:
```

Comment:
can be written label = new_label or item.label

Reply:
Done